tests-legacy/Unit/Classes/ToolsCoreTest.php

```diff
@@ -369,6 +369,48 @@ public function testStrReplaceFirst($search, $replace, $subject, $cur, $expected
         $this->assertEquals($expected, Tools::StrReplaceFirst($search, $replace, $subject, $cur));
     }
 
+    /**
+     * @param string $passwordGenerated
+     * @param string $expectedPassword
+     *
+     * @dataProvider passwodGenProvider
+     */
+    public function testPasswdGen($expectedPassword, $passwordGenerated)
```

Comment:
```suggestion
    public function testPasswdGen(string $expectedPassword, string $passwordGenerated): void
```

Reply:
thanks @Progi1984, but passwordGen can retourn string, int or bool (false)

Reply:
Type hint the first parameter in this case.